### ISSUE TITLE
Volatility introduced in tests since approx September 18th - potentially package sync

### ISSUE THREAD

**Issue by ncclementi** [coiled/coiled-runtime#446]
Some tests that were decently stable have become very volatile around mid september. This makes it very hard to trust the regression detection system we have in place. But there is a pattern:

q3 [5GB parquet] - upstream 
![Screen Shot 2022-10-13 at 12 51 11 PM](https://user-images.githubusercontent.com/7526622/195657486-0d85b657-1c0f-4a87-bc35-20d18eb4a34a.png)

q5 [5GB parquet] - upstream - Hard to see but the red line became more volatile around the same time
![Screen Shot 2022-10-13 at 12 54 52 PM](https://user-images.githubusercontent.com/7526622/195658284-77b9620a-d277-4e6a-bc74-62da07344121.png)

q7 [5GB parquet] - upstream 
![Screen Shot 2022-10-13 at 12 56 21 PM](https://user-images.githubusercontent.com/7526622/195658591-d2a4f96f-a034-435c-a7d5-655a4b0b74bb.png)

q8 [5GB parquet] - upstream 
![Screen Shot 2022-10-13 at 12 56 56 PM](https://user-images.githubusercontent.com/7526622/195658706-21adbda3-7db6-425f-b7e6-c6631fb72550.png)

test_dataframe_align - volatile and definitely a regression
![Screen Shot 2022-10-13 at 12 58 17 PM](https://user-images.githubusercontent.com/7526622/195658974-a0c69872-4eb1-49b7-b367-6fddd187eaad.png)

test_shuffle - became more volatile
![Screen Shot 2022-10-13 at 12 59 26 PM](https://user-images.githubusercontent.com/7526622/195659198-2afc8de3-644b-418c-8b5f-f733e705336e.png)

The interesting thing is that this volatility also happens on latest (2022.6.1) which makes me think this issue was introduced with something that happened in package sync (PR merged 09/15) 

q3 [5GB parquet] latest
![Screen Shot 2022-10-13 at 1 04 51 PM](https://user-images.githubusercontent.com/7526622/195660411-93e6c4e4-41a6-4caa-8479-cf8a385cdade.png)

q7 [5GB parquet] latest
![Screen Shot 2022-10-13 at 1 06 04 PM](https://user-images.githubusercontent.com/7526622/195660856-3b115480-b304-43a6-9342-4937d006697f.png)

test_dataframe_align
![Screen Shot 2022-10-13 at 1 06 55 PM](https://user-images.githubusercontent.com/7526622/195661096-36d1681d-7649-455a-9f43-496e03a29af3.png)

cc: @ian-r-rose @jrbourbeau @shughes-uk

**Comment by ncclementi** [comment]
@fjetter and @ntabris for visibility

**Comment by fjetter** [comment]
Comparing conda environments between pre and post merge, the most interesting diff is coiled itself which jumped from 0.2.27 to 0.2.30 but I only compared two runs. I'd be interested to see a comparison of more commits, particularly the spikes.
Do we have an easy way checking what versions where installed on all the runs? Is this something we should consider adding to the benchmark database, e.g. output of client.get_versions?

**Comment by ntabris** [comment]
> Do we have an easy way checking what versions where installed on all the runs? Is this something we should consider adding to the benchmark database, e.g. output of client.get_versions?

Two thoughts.

1. I'm interesting in having better tagging/tracking on the platform side, rather than a thing that's specifically for coiled-runtime tests.
2. I'll take a quick look and see if I can get some data about version changes around Sept 19.

**Comment by ntabris** [comment]
It's very hard to find data for old tests. So here's what I can get without ridiculous amounts of work.

Here's a `test_array` cluster from **after** volatility increased:

https://cloud.coiled.io/dask-engineering/clusters/83375/details

That cluster has py3.9 and dask/distributed 2022.6.0. You can follow that link to see other software versions.

Here's cluster from **before** that has py3.9 and dask/distributed 2022.6.0:

https://cloud.coiled.io/dask-engineering/clusters/75547/details

I see changes in pandas and pyarrow. Would that be relevant? Anyone else is welcome to dig in more comparing those clusters, I don't have much more insight.

**Comment by shughes-uk** [comment]
I've identified a couple of issues with package sync, in both detecting packages on cluster and client side. Staging has a ton of bugfixes i'm eager to get out but it's held up currently by ensuring we've tested some of the infra work.

**Comment by ntabris** [comment]
> I've identified a couple of issues with package sync, in both detecting packages on cluster and client side. Staging has a ton of bugfixes i'm eager to get out but it's held up currently by ensuring we've tested some of the infra work.

Any reason to think package sync bugs are causing this?

**Comment by shughes-uk** [comment]
I suspect it might be causing release versions to be installed instead of main branch. If this is happening even on latest, then it's unlikely.

**Comment by ntabris** [comment]
> I suspect it might be causing release versions to be installed instead of main branch. If this is happening even on legacy tests, then it's unlikely.

Yeah, you can see the change on clusters running `coiled-runtime=0.1.0`: https://benchmarks.coiled.io/coiled-0.1.0-py3.9.html

The examples I listed https://github.com/coiled/coiled-runtime/issues/446#issuecomment-1282769796 are runtime 0.1.0 w/ py39.

**Comment by ncclementi** [comment]
@shughes-uk Should we have an "environment" in the runtime where we test against staging? 
It would dask/distributed upstream + coiled staging. Or is this something y'all have covered on your end?

**Comment by shughes-uk** [comment]
If we can identify a subsection of the tests for this it might be worthwhile, otherwise the cost might be prohibitive

**Comment by ncclementi** [comment]
It seems most of the volatility has calm down, except for the case of `test_dataframe_align.` I would be inclined to close this issue and talk about dataframe align in a separate one, thoughts?

q3 [5GB parquet] - upstream
![Screen Shot 2022-10-31 at 12 50 21 PM](https://user-images.githubusercontent.com/7526622/199063604-50fc962b-40dd-463c-b03c-9d6f29167f72.png)

q5 [5GB parquet] - upstream
![Screen Shot 2022-10-31 at 12 51 04 PM](https://user-images.githubusercontent.com/7526622/199063749-f44fae46-2fd2-4cf4-b080-38174ce78043.png)

q7 [5GB parquet] - upstream
![Screen Shot 2022-10-31 at 12 51 47 PM](https://user-images.githubusercontent.com/7526622/199063907-d582fa44-edf9-4158-b90f-a1b0287bc8ee.png)

q8 [5GB parquet - upstream
![Screen Shot 2022-10-31 at 12 52 33 PM](https://user-images.githubusercontent.com/7526622/199064071-ab838708-f5d6-4a18-9622-28c9524b2361.png)

test_dataframe_align - upstream (still crazy jumps)
![Screen Shot 2022-10-31 at 12 53 43 PM](https://user-images.githubusercontent.com/7526622/199064267-3116f816-b798-4cd6-ac39-7461b0428f56.png)

test_shuffle - upstream
![Screen Shot 2022-10-31 at 12 54 37 PM](https://user-images.githubusercontent.com/7526622/199064439-c4dcc22d-287e-4417-aaaa-b616abdf8b65.png)

q3 [5GB parquet] latest
![Screen Shot 2022-10-31 at 12 55 49 PM](https://user-images.githubusercontent.com/7526622/199064686-406961ec-b925-4615-aaf7-6f31093df82f.png)

q7 [5GB parquet] latest
![Screen Shot 2022-10-31 at 12 56 44 PM](https://user-images.githubusercontent.com/7526622/199064878-8b4de0ac-2009-4e5f-8668-fc942a41754c.png)

 test_dataframe_align  latest
![Screen Shot 2022-10-31 at 12 57 08 PM](https://user-images.githubusercontent.com/7526622/199064955-0c585a79-ee1f-4f7d-8975-d78dc0775d07.png)

**Comment by ntabris** [comment]
vorticity also shows non-trivial variability

**Comment by ncclementi** [comment]
> vorticity also shows non-trivial variability

Yes, but that has been that way since the beginning, maybe @gjoseph92 knows more about it.

![Screen Shot 2022-10-31 at 1 07 41 PM](https://user-images.githubusercontent.com/7526622/199067032-6de9df9c-2845-4ebf-b166-587962ee2178.png)

**Comment by gjoseph92** [comment]
I don't think that's too surprising, I think that one spills a bit? I'd expect it to get better after https://github.com/dask/distributed/issues/7213

**Comment by ntabris** [comment]
> I don't think that's too surprising, I think that one spills a bit?

Yep. Spill is also why `test_dataframe_align` sometimes is quick (when no spill) and sometimes slow (when lots of spill).

**Comment by gjoseph92** [comment]
This could even be the same thing as `test_dataframe_align`: if work stealing decides to mess with the co-assignment, that could also affect this test I think
